lib/nolibc/stdlib.c

```diff
@@ -61,6 +61,60 @@
 
 #define __DECONST(type, var) ((type)(uintptr_t)(const void *)(var))
 
+
+#if CONFIG_LIBUKALLOC
+/* Allocate size bytes of memory. Returns pointer to start of allocated memory,
+ * or NULL on failure.
+ */
+void *malloc(size_t size)
+{
+	return uk_malloc(uk_alloc_get_default(), size);
+}
+
+/* Release memory previously allocated by malloc(). ptr must be a pointer
+ * previously returned by malloc(), otherwise undefined behavior will occur.
+ */
+void free(void *ptr)
+{
+	return uk_free(uk_alloc_get_default(), ptr);
+}
+
+/* Allocate memory for an array of nmemb elements of size bytes. Returns
+ * pointer to start of allocated memory, or NULL on failure.
+ */
+void *calloc(size_t nmemb, size_t size)
+{
+	return uk_calloc(uk_alloc_get_default(), nmemb, size);
+}
+
+/* Change the size of the memory block pointed to by ptr to size bytes.
```

Comment:
I think we may specify that the allocated memory remains unchanged if the call fails